### PR TITLE
Make readonly (for complex value types) (Fix #808) 

### LINKED ISSUE #808: Make readonly (for complex value types)
Make complex value type readonly.

Before:

``` csharp
public class MyClass
{
    private MyStruct myStruct = default(MyStruct);
}
private struct MyStruct
{
    public int Value;
}
```

After:

``` csharp
public class MyClass
{
    private readonly MyStruct myStruct = default(MyStruct);
}
private struct MyStruct
{
    public int Value;
}
```

Diagnostic Id: `CC0121`
Category: `Usage`
Severity: `Info`

This analyzer should be disabled by default because of the problems described on #775. And it should share could, if possible, with the analyzer for `CC0052`. The code fix can be the same, for sure, and should only be updated to take `CC0121` as an extra id.

### PR DESCRIPTION
**Pull request by cezarant**
My second attempt contribution. 
In this pull request i'm putting the necessary files to resolve the issue #808  
I hope that the changes have been sufficient.

### CODE REVIEW

**Comment by cezarant** [comment]
Do you have any suggestions to improve my test coverage? It is normal to have this percentage in the first pull's?

**Comment by carloscds** [comment]
@cezarant If possible, write tests for many scenarios of your analyzer and fixer,

File: src/CSharp/CodeCracker/Usage/ReadOnlyComplexTypesAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+using System.Linq;
+using System.Collections.Generic;
+using System;
+
+namespace CodeCracker.CSharp.Usage
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class ReadOnlyComplexTypesAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string Title = "Complex files readonly";
```
Comment (carloscds) [inline src/CSharp/CodeCracker/Usage/ReadOnlyComplexTypesAnalyzer.cs]: @cezarant Change this text to "Change complex fields to readonly"

File: src/CSharp/CodeCracker/Usage/ReadOnlyComplexTypesAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        internal const string Message = "Make '{0}' readonly";
+        internal const string Category = SupportedCategories.Usage;
+        const string Description = "Complex files must be readonly.";
```
Comment (carloscds) [inline src/CSharp/CodeCracker/Usage/ReadOnlyComplexTypesAnalyzer.cs]: @cezarant Change to "Complex fields must be readonly."

File: src/CSharp/CodeCracker/Usage/ReadOnlyComplexTypesAnalyzer.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+        internal static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId.ReadOnlyComplexTypes.ToDiagnosticId(),
+            Title,
+            Message,
+            Category,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: false,
```
Comment (carloscds) [inline src/CSharp/CodeCracker/Usage/ReadOnlyComplexTypesAnalyzer.cs]: @cezarant Why this analyzer is disable ?

Comment (cezarant) [inline src/CSharp/CodeCracker/Usage/ReadOnlyComplexTypesAnalyzer.cs]: @carloscds  Because the problems reported in #775

**Comment by carloscds** [comment]
@cezarant  Do you have some news about this ?

**Comment by cezarant** [comment]
Hello @carloscds I will work on it today .

**Pull request closed without merging**